### PR TITLE
Resolve circular dependencies for project org.apache.maven:maven-xml-impl

### PR DESCRIPTION
**Pull request by w6et**
… resolve dependencies for project org.apache.maven:maven-xml-impl

reason:Circular dependencies：
maven-xml-impl
    ->plexus-xml
        ->maven-xml-impl
plexus-xml need to exclusion maven-xml-impl
2)rename 'Implementation of Maven API XML' to 'maven-xml-impl(Implementation of Maven API XML)'

Following this checklist to help us incorporate your
contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue. Your pull request should address just this issue, without
       pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MNG-XXX] SUMMARY`,
       where you replace `MNG-XXX` and `SUMMARY` with the appropriate JIRA issue.
 - [ ] Also format the first line of the commit message like `[MNG-XXX] SUMMARY`.
       Best practice is to use the JIRA issue title in both the pull request title and in the first line of the commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

File: maven-xml-impl/pom.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@ under the License.
   </parent>
 
   <artifactId>maven-xml-impl</artifactId>
-  <name>Implementation of Maven API XML</name>
+  <name>maven-xml-impl(Implementation of Maven API XML)</name>
```
Comment (gnodet) [inline maven-xml-impl/pom.xml:29]: This is not coherent with the other packages afaik.
I'd rather do that globally all all modules if we want to go that way.

File: maven-xml-impl/pom.xml [hunk 2]
```diff
@@ -48,6 +48,12 @@ under the License.
       <artifactId>plexus-xml</artifactId>
       <version>${plexusXmlVersion}</version>
       <scope>provided</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>org.apache.maven</groupId>
+          <artifactId>maven-xml-impl</artifactId>
+        </exclusion>
+      </exclusions>
```
Comment (gnodet) [inline maven-xml-impl/pom.xml:56]: Doesn't the current module take precedence during resolution ?
What kind of issues does it actually cause ?

Comment (w6et) [reply maven-xml-impl/pom.xml:56]: > Doesn't the current module take precedence during resolution ? What kind of issues does it actually cause ?
in  my landtop，windows11 ,apache-maven-3.9.4
run command 'mvn clean compile  -Dmaven.test.skip=true ' in maven home_dir(my case is D:\works\git\maven),it occured an exception as below:
[ERROR] Failed to execute goal on project maven-xml-impl: Could not resolve dependencies for project org.apache.maven:maven-xml-impl:jar:4.0.0-alpha-8-SNAPSHOT: The following artifacts could not be resolved: org.apache.maven:maven-xml-impl:jar:4.0.0-alpha-8-SNAPSHOT (absent): org.apache.maven:maven-xml-impl:jar:4.0.0-alpha-8-SNAPSHOT was not found in https://xxxxxxxxxx/repository/public during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of aliyun has elapsed or updates are forced -> [Help 1]
.......
**It looks like maven-xml-impl is needed to get plexus-xml, but maven-xml-impl is being built**

Comment (gnodet) [reply maven-xml-impl/pom.xml:56]: I wonder if that's a bug somehow in the resolution... @cstamas ?

**Comment by hboutemy** [comment]
I agree that there is a problem
I don't think the current PR is the right approach
as found [on the ML while discussing on the topic](https://lists.apache.org/thread/8t5szmn8fhlrklhg5dgv496k26tz3w3p), I thin the approach is to create a Maven Shared Component that provides MetaAnnotations, XML API and XML Implementation separately from Maven core
I don't know how to name this release: "Maven fundamentals"?

**Comment by gnodet** [comment]
> I agree that there is a problem
> I don't think the current PR is the right approach
> as found [on the ML while discussing on the topic](https://lists.apache.org/thread/8t5szmn8fhlrklhg5dgv496k26tz3w3p), I thin the approach is to create a Maven Shared Component that provides MetaAnnotations, XML API and XML Implementation separately from Maven core
> I don't know how to name this release: "Maven fundamentals"?

While I'm not opposed to moving things, I don't think this will solve the problem.
The problem here is the cycle between maven-xml-impl / plexus-xml / maven-xml-impl.
Moving maven-xml-impl in another location won't change that problem, so while this may not be a sufficient solution, I think this PR is still needed.

**Review by gnodet — CHANGES_REQUESTED** [review]
@w6et Could you please remove the module name change ?

**Comment by w6et** [comment]
> @w6et Could you please remove the module name change ?

Done, @gnodet

**Review by gnodet — APPROVED** [review]

**Pull request merged**